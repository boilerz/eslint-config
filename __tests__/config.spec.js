const backendConfig = require('../index');
const frontendConfig = require('../frontend');

describe('config', () => {
  it('should handle backend and frontend configs', () => {
    expect(backendConfig).toMatchSnapshot();
    expect(frontendConfig).toMatchSnapshot();
  });
});
