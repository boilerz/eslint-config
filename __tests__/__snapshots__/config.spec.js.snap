// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config should handle backend and frontend configs 1`] = `
Object {
  "env": Object {
    "jest": true,
  },
  "extends": Array [
    "airbnb/base",
    "plugin:@typescript-eslint/recommended",
    "prettier/@typescript-eslint",
    "plugin:prettier/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": Object {
    "ecmaVersion": 2018,
    "sourceType": "module",
  },
  "plugins": Array [
    "@typescript-eslint",
  ],
  "rules": Object {
    "@typescript-eslint/ban-ts-ignore": 0,
    "import/extensions": 0,
  },
  "settings": Object {
    "import/order": Array [
      "error",
      Object {
        "alphabetize": Object {
          "caseInsensitive": true,
          "order": "asc",
        },
        "groups": Array [
          "builtin",
          "external",
          "internal",
        ],
        "newlines-between": "always",
        "pathGroups": Array [
          Object {
            "group": "internal",
            "pattern": "@boilerz/**",
            "position": "after",
          },
        ],
        "pathGroupsExcludedImportTypes": Array [
          "builtin",
        ],
      },
    ],
    "import/resolver": Object {
      "node": Object {
        "extensions": Array [
          ".js",
          ".jsx",
          ".ts",
          ".tsx",
        ],
      },
    },
  },
}
`;

exports[`config should handle backend and frontend configs 2`] = `
Object {
  "env": Object {
    "jest": true,
  },
  "extends": Array [
    "airbnb",
    "plugin:@typescript-eslint/recommended",
    "prettier/@typescript-eslint",
    "plugin:prettier/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": Object {
    "ecmaVersion": 2018,
    "sourceType": "module",
  },
  "plugins": Array [
    "@typescript-eslint",
  ],
  "rules": Object {
    "@typescript-eslint/ban-ts-ignore": 0,
    "import/extensions": 0,
    "react/jsx-filename-extension": 0,
  },
  "settings": Object {
    "import/order": Array [
      "error",
      Object {
        "alphabetize": Object {
          "caseInsensitive": true,
          "order": "asc",
        },
        "groups": Array [
          "builtin",
          "external",
          "internal",
        ],
        "newlines-between": "always",
        "pathGroups": Array [
          Object {
            "group": "internal",
            "pattern": "@boilerz/**",
            "position": "after",
          },
        ],
        "pathGroupsExcludedImportTypes": Array [
          "builtin",
        ],
      },
    ],
    "import/resolver": Object {
      "node": Object {
        "extensions": Array [
          ".js",
          ".jsx",
          ".ts",
          ".tsx",
        ],
      },
    },
  },
}
`;
